content/guides/http/go/net_http/server.md

```diff
@@ -0,0 +1,231 @@
+---
+title: "Server"
+date: 2018-11-27T01:12:31-07:00
+draft: false
+aliases: [/integrations/http/go/net_http/server]
+weight: 2
+logo: /images/gopher.png
+---
+
+- [Introduction](#introduction)
+- [Traces](#traces)
+- [Metrics](#metrics)
+- [End-to-end example](#end-to-end-example)
+- [Viewing traces](#view-traces)
+- [Viewing metrics](#view-metrics)
+- [References](#references)
+
+### Introduction
+
+OpenCensus provides a package [go.opencensus.io/plugin/ochttp](https://godoc.org/go.opencensus.io/plugin/ochttp) which has a custom HTTP handler [ochttp.Handler](https://godoc.org/go.opencensus.io/plugin/ochttp#Handler) that can wrap your [HTTP handlers](https://golang.org/pkg/net/http/#Handler) while running your [HTTP servers](https://golang.org/pkg/net/http/#Server). For example
+
+```go
+package main
+
+import (
+	"log"
+	"net/http"
+
+	"go.opencensus.io/plugin/ochttp"
+)
+
+func main() {
+	originalHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("Hello, World!"))
+	})
+	och := &ochttp.Handler{
+		Handler: originalHandler, // The handler you'd have used originally
+	}
+
+	// Now use the instrumnted handler
+	if err := http.ListenAndServe(":9999", och); err != nil {
+		log.Fatalf("Failed to run the server: %v", err)
+	}
+}
+```
+
+### Traces
+The instrumented handler creates a trace automatically for each request that is received. We shall need a [Trace exporter](/exporters/supported-exporters/go), allowing us to examine the traces.
+If the request that it received contains a trace with the same context propagation format, it will continue the trace as a root, for example
+
+![](/images/ochttp-server-trace-non-link.png)
+
+However, if the handler is started with the field [IsPublicEndpoint](https://godoc.org/go.opencensus.io/plugin/ochttp#Handler.IsPublicEndpoint), then the span of the propagated span will be recorded as a [Parent Link](/tracing/span/link/) instead for example
+
+On the server side
+![](/images/ochttp-server-trace-link-server.png)
+
+which is the parent link of
+![](/images/ochttp-server-trace-link-client.png)
```

Comment:
two images do not show how they are linked.

Reply:
Kinda, see this "The child" is shown then the parent
<img width="836" alt="screen shot 2018-12-01 at 4 06 09 am" src="https://user-images.githubusercontent.com/4898263/49328045-770d8980-f51e-11e8-876a-b6ba0eb15fac.png">

Perhaps AWS X-Ray can help out with the linking graph.

Reply:
I see the child traceId in parent. But the child doesn't show any id so it his hard to correlate. I would be nice to have some correlating info but I understand it may not be possible.